pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -409,9 +409,16 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 			changed = true
 			return changed
 		}
-		if remaining < uint64(config.FixedResources.Memory)<<10 {
-			errStr := fmt.Sprintf("Remaining memory bytes %d app instance needs %d\n",
-				remaining, config.FixedResources.Memory<<10)
+		need := uint64(config.FixedResources.Memory) << 10
+		if remaining < need {
+			var errStr string
+			if remaining+halting < need {
+				errStr = fmt.Sprintf("Remaining memory bytes %d app instance needs %d",
+					remaining, need)
+			} else {
+				errStr = fmt.Sprintf("App instance needs %d bytes; waiting for halting app instances to free up %d bytes",
+					need, halting)
```

Comment:
Can we also add here remaining memory bytes to keep it visible from error?
Do we need to set `MissingMemory` flag in case of `remaining+halting < need`?

Reply:
I'll update the log.
Code already sets MissingMemory on line419.

Reply:
> Code already sets MissingMemory on line419.

But should it set MissingMemory in both cases? Or your changes here mostly about logging, not the logic?

Reply:
If is after the if/then/else I added, so it does apply independently of the log message. Does your screen show something different?